packages/db/src/models/base.ts

```diff
@@ -17,6 +17,23 @@ class BasePgModel<ItemType, RecordType extends { cumulus_id: number }> {
     this.tableName = tableName;
   }
 
+  async count(
+    knexOrTransaction: Knex | Knex.Transaction,
+    params: ([string, string, string] | [Partial<RecordType>])[]
+  ) {
+    const query = knexOrTransaction(this.tableName).where((builder) => {
+      params.forEach((param) => {
+        if (param.length === 3) {
+          builder.where(...param);
+        }
+        if (param.length === 1) {
+          builder.where(param[0]);
```

Comment:
These seem like two very specific use cases that I don't quite understand, despite the test cases below. 🤔

Reply:
Hmm, fair enough.  This might also be relying a bit too much on the types to protect us.  

The use cases are in the typing: 

`params: ([string, string, string] | [Partial<RecordType>])[]` 

Which is to say, it's an array of either query objects like 

['foo', '>', 'bar'] or [{record object}]

Also docstring might help.     The intent here is to, given a table, query a count of objects that match a partial record *or* a specific compactor query.

Reply:
OK, I guess it's just a convention of the `where` function to have both formats. Seems fine, but are we sure we want to expose that further into our own code? I understand we are already doing this in a few places and don't expect you to change that, but wanted to ask regardless.

Reply:
After checking the `knex` docs, I guess the `[string, string, string]` variable is how you pass in operators? I'm fine with this.

Reply:
Oh, I see -- yes that's what Knex expects.